solc_json_parser/parser.py

```diff
@@ -988,6 +988,82 @@ def get_deploy_bin_by_hash(self, hsh: str) -> Optional[str]:
         '''Get deployment binary by hash of fully qualified contract / library name'''
         return self.get_any(self.qualified_name_from_hash(hsh), 'bin')
 
+    def get_literals(self, contract_name: str, only_value=False) -> dict:
+        """
+        Get all literals(number, address, string, other) in the contract.
+        for 'other' type, if only_value is True, return the string value
+        - `contract_name`: contract_name in string
+        - `only_value`: set to true to get only values, otherwise get all literal objects
+        """
+
+        # traverse the dictionary and get all the literals
+        literals_nodes = set()
+
+        def traverse(node):
```

Comment:
it might be easier to read if this function is extracted to a method if it reads the instance state, or a function if it does not read the object state.

Reply:
I have refractor the code and pushed to github